### PR TITLE
Responsive UI fixes

### PR DESCRIPTION
**Pull request by fbarl**
A couple of surgical fixes to the responsive aspect of the UI based on some recent issues / comments.

#### Changes

* Fix left side panel overflow on mobile - addresses https://github.com/tmrowco/electricitymap-contrib/commit/58a5d495125fecf0a73fd236d1cd906afd54067f#r40174062
* Look at screen width instead of isMobile for `/info` page redirect - the new `useIsSmallScreen` / `useIsMediumUpScreen` React hooks can be used instead of CSS screen size breakpoints when more of the styles is moved to React components #2261
* Update exchange arrows under window width resizing - fixes https://github.com/tmrowco/electricitymap-contrib/issues/2573

### CODE REVIEW

File: web/src/layout/map.js [hunk 1]
```diff
@@ -146,16 +147,37 @@ export default () => {
...
 
+  // Update the map viewport whenever the window width or height changes.
+  // TODO: This is a hack that should ideally not be needed, but without it
+  // the exchange arrows don't seem to update well under resizing the width.
+  // See https://github.com/tmrowco/electricitymap-contrib/issues/2573#issuecomment-658242753
+  const { width, height } = useWindowSize();
+  useEffect(() => {
+    handleViewportChange({ ...viewport, width, height });
+  }, [width, height]);
+
```
Comment (Kongkille) [inline web/src/layout/map.js:180]: Do you know if this affects performance? I know we have had a lot of issues with the exchange arrows and the performance in the past.

Even still this is probably still the best solution.

Just curious, in the bug found by q--, it seemed to only occur when dragging horizontally, that is when the width changes and the height stay constant. Do we then need to use this "hack" for both height and width changes, or could we just use it for width changes only?

Comment (fbarl) [reply web/src/layout/map.js:180]: > Do you know if this affects performance?

I tested it quickly and it seems like the answer is not really. Also, I think it's important to note that window resizing doesn't happen that often and the performance in fixed viewport should be completely unaffected by this code.

> Do we then need to use this "hack" for both height and width changes, or could we just use it for width changes only?

You're right, in this particular case I think we can remove the height listener but I decided to keep it as I wasn't entirely sure if not tracking the width is a bug or rather tracking the height is an accidental _feature_ so I added both here to be on the safe side regarding the future changes :slightly_smiling_face:

Comment (Kongkille) [reply web/src/layout/map.js:180]: Good points, thanks :)

File: web/src/hooks/viewport.js [hunk 2]
```diff
@@ -45,3 +45,36 @@ export function useHeightObserver(ref, offset = 0) {
...
+
+export function useIsSmallScreen() {
+  const { width } = useWindowSize();
+  return width < 768;
+}
+
+export function useIsMediumUpScreen() {
+  const { width } = useWindowSize();
+  return width >= 768;
+}
```
Comment (FelixDQ) [inline web/src/hooks/viewport.js:80]: NIce hooks :-)

Comment (fbarl) [reply web/src/hooks/viewport.js:80]: Thanks :wink: And thank you for the review! :pray:

**Pull request merged**